pkg/serviceaccount/issuermetadata.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package serviceaccount
+
+import (
+	"encoding/json"
+	"fmt"
+
+	jose "gopkg.in/square/go-jose.v2"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// issuerMetadata provides a minimal subset of OIDC provider metadata:
+// https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata
+type issuerMetadata struct {
+	Issuer string `json:"issuer"` // REQUIRED in OIDC; meaningful to relying parties.
+	// AuthzEndpoint string   `json:"authorization_endpoint"`                // REQUIRED in OIDC; but useless to relying parties.
```

Comment:
still outstanding... I would not expect to serve a non-spec compliant discovery doc

Reply:
Our goal should be discovery compatibility with relying parties that need to validate ID tokens. Relying parties will not be initiating login flows. Given that I don't know what to put in it and there's no RFC police to arrest us for not being compliant to the tee, I'd leave it out until someone files a bug.

Reply:
![rfc-not-a-cop](https://user-images.githubusercontent.com/980082/72916319-d7a5ea00-3d0f-11ea-9c67-1df5cafb2475.png)

Reply:
SG. Added a comment to that effect.